### DISCUSSION TITLE
access network interface

### DISCUSSION THREAD

**Discussion by rigorvzla** [LibreHardwareMonitor/LibreHardwareMonitor#1437] [General]
Hello everyone, I have a big problem, and is that I can not find how to access the network interface by any means, as it generates an accessibility error, when I want to access by the code that I have shown you saying that it has access problems, I do not know if it is a library error or misuse on my part, I hope you can help me.

```
 foreach (IHardware hardware in computer.Hardware)
 {
     if (hardware.HardwareType == HardwareType.Network)
     {
       var speed = ((NetworkInterface)((LibreHardwareMonitor.Hardware.Network.)hardware).NetworkInterface).Speed;
...
...
```

**Comment by MarkVabulas** [comment]
You need to run your main software as Administrator.

**Reply by rigorvzla** [reply]
yes i do it, but cant acces , LibreHardwareMonitor.Hardware.Network. not is class public.
